p2p/net/conn.go

```diff
@@ -157,12 +164,16 @@ func (c *FormattedConnection) publish(message []byte) {
 // Concurrency: can be called from any go routine
 func (c *FormattedConnection) Send(m []byte) error {
 	c.wmtx.Lock()
+	c.deadliner.SetReadDeadline(time.Now().Add(c.deadline))
```

Comment:
on send we want to use the WriteDeadline, though I'm taking care of this.

Reply:
we can close this pr if you prefer, i just wanted you to have what we found